src/devices/Nmea0183/Sentences/HeadingAndDeclination.cs

```diff
@@ -0,0 +1,191 @@
+                }
+            }
+        }
+
+        /// <summary>
+        /// This is true for this message type
+        /// </summary>
+        public override bool ReplacesOlderInstance => true;
+
+        /// <summary>
+        /// Angle of the wind
+        /// </summary>
+        public Angle HeadingTrue
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// Magnetic heading (derived from true heading and declination)
+        /// </summary>
+        public Angle? HeadingMagnetic
+        {
+            get
+            {
+                if (Declination.HasValue)
+                {
+                    return (HeadingTrue - Declination.Value).Normalize(true);
+                }
+
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Deviation at current location. Usually unknown (empty). Not sure what this field means. The deviation is a property
+        /// of the actual compass and the current orientation, not the location.
+        /// </summary>
+        public Angle? Deviation
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// Magnetic declination (sometimes also called variation) at current location. Usually derived from the NOAA magnetic field model by one
+        /// of the attached devices.
+        /// </summary>
+        public Angle? Declination
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// Presents this message as output
+        /// </summary>
+        public override string ToNmeaParameterList()
+        {
+            if (Valid)
+            {
+                StringBuilder b = new StringBuilder();
+                b.AppendFormat(CultureInfo.InvariantCulture, "{0:F1},", HeadingTrue.Normalize(true).Degrees);
+                if (Deviation.HasValue)
+                {
+                    b.AppendFormat(CultureInfo.InvariantCulture, "{0:F1},{1},", Math.Abs(Deviation.Value.Degrees),
+                        Deviation.Value.Degrees >= 0 ? "E" : "W");
+                }
+                else
+                {
+                    b.AppendFormat(",,");
+                }
+
+                if (Declination.HasValue)
+                {
+                    b.AppendFormat(CultureInfo.InvariantCulture, "{0:F1},{1}", Math.Abs(Declination.Value.Degrees),
+                        Declination.Value.Degrees >= 0 ? "E" : "W");
+                }
+                else
+                {
+                    b.AppendFormat(",");
+                }
+
+                return b.ToString();
+            }
+
+            return string.Empty;
+        }
+
+        /// <inheritdoc />
+        public override string ToReadableContent()
+        {
+            if (Valid && Declination.HasValue)
+            {
+                return $"Magnetic Heading: {HeadingTrue.Degrees:F1}°, Declination: {Declination.Value.Degrees:F1}°";
```

Comment:
the ° should be encoded as unicode like we do in the samples.

Reply:
I've not done this for a long time. Our code files are all encoded in UTF8-BOM, so this should give no problems at all (otherwise, we would be in big trouble with files such as https://github.com/dotnet/iot/blob/main/src/devices/Common/Iot/Device/Graphics/LcdCharacterEncodingFactory.cs)